CaC Runbooks blog (#1196)
OctopusDeploy/blog: blog/2025-02/cac-runbooks/index.md

---
title: Introducing Config as Code for Runbooks

> description: Everything you need to know about the new config as code for Runbooks

Reviewer: ```suggestion
description: Everything you need to know about the new Config as Code for Runbooks
```

author: harriet.alexander@octopus.com
visibility: private
published: 2025-02-17-1400
metaImage: cacrunbooks image.png
bannerImage: cacrunbooks image.png

> bannerImageAlt: Runbook within code syntax.

Reviewer: ```suggestion
bannerImageAlt: Graphic of a woman sitting on a large stack of books while working on laptop, surrounded by large, 3 dimensional curly braces
```

isFeatured: false
tags: 
  - product, Configuration as Code, Runbooks
---

### Introducing Config as Code for Runbooks

> We are excited to announce a feature that many have anticipated for a long time: Config as Code for Runbooks. This feature adds Config as Code support to Runbooks it gives you a seamless, version-controlled way to manage your workflows. This new addition lets you run, create and track your Runbooks in your chosen Git respository bringing your runbooks together with your Deployments process. This means less places to make changes and provides more control over your workflows. 

Reviewer: ```suggestion
We are excited to announce a feature that many have anticipated for a long time: Config as Code for Runbooks. This feature adds Config as Code support to Runbooks it gives you a seamless, version-controlled way to manage your workflows. This new addition lets you run, create, and track your Runbooks in your chosen Git repository bringing your runbooks together with your deployment process. This means fewer places to make changes and provides more control over your workflows. 
```

* Familiar Workflow: If you're used to version-controlled projects, you'll find Config as Code for Runbooks natural. Following Git principles will make the process straightforward.

### Migrating Existing Runbooks
If you use Octopus's Runbooks, we have a tool to help you migrate to Config as Code for Runbooks.
* Migration Process: When you start the migration, Octopus will move your published Runbooks to a new folder, `/runbooks`, in your chosen repository. This folder will store the published snapshot and configuration from the Runbook.
* We will migrate any draft versions of your Runbook to a separate sub-folder `/runbooks/migrated-draft`. You won't see these in the UI. But you can access them by moving them into the main Runbooks folder using a pull request (PR).
* History Retention: Don't worry about losing your history. The UI will still show your existing Runbook runs.

> ![CaC Runbook Migration] (2025-02-11 08.07.28.gif) 

Reviewer: ```suggestion
![Screen recording of Runbooks migration process. Starting on the Runbook list, the user opens the wizard, reviews the changes, creates a new branch named feat/migrate-runbooks, uses the default commit message, reviews the migration and completes it. Once completed, the Runbooks list screen is automatically refreshed showing a branch selector and an alert showing there are un-migrated drafts] (2025-02-11 08.07.28.gif) 
```
